Hot-Rod-Transactions-NonXA.asciidoc

```diff
@@ -0,0 +1,311 @@
+Lightweight transactions for Hot Rod
+====================================
+
+*JIRA*: https://issues.jboss.org/browse/ISPN-7807[ISPN-7807]
+
+== Introduction
+
+This document is a design document for lightweight Hot Rod transaction and Synchronization enlistment. Full XA support will be described in another document.
+
+The transaction state is kept in the client during the transaction execution and the server is only aware of the transaction when the client wants to commit.
+
+== Changes to existing code
+
+Since a default `TransactionManager` is needed for the client, a base implementation is extracted from the `EmbeddedTransactionManager` to commons package.
+The `Transaction`'s implementation (and it dependencies) will be moved to commons package.
+
+== Limitation and Requirements
+
+* In the client, the transaction will be enlisted as `Synchronization`.
+XA support will be added later in another document.
+
+* The server will use `REPEATABLE_READ` isolation.
+
+* The server will use the `EmbeddedTransactionManager`.
+It needs to invoke the prepare and commit/rollback phase individually.
+
+* Recovery needs to be enabled in the server.
+It prevents the transaction to be automatically rolled-back when the originated leaves the cluster.
```

Comment:
Recovery requires manual intervention to clean up locks and a lot of other baggage (i.e. XIDs). I'd rather add a flag to client transactions so that the TransactionTable cleanup doesn't remove them when the originating node fails.